tsl/src/continuous_aggs/materialize.c

```diff
@@ -495,12 +522,42 @@ scan_take_invalidation_tuple(TupleInfo *ti, void *data)
 	return SCAN_CONTINUE;
 }
 
+/* use materialization_invalidation_threshold to filter the logs */
+static ScanTupleResult
+scan_del_materialization_invalidation_tuple(TupleInfo *ti, void *data)
+{
+	InvalidationScanState *scan_state = (InvalidationScanState *) data;
+
+	MemoryContext old_ctx = MemoryContextSwitchTo(scan_state->mctx);
+	Form_continuous_aggs_materialization_invalidation_log invalidation_form =
+		((Form_continuous_aggs_materialization_invalidation_log) GETSTRUCT(ti->tuple));
+	Invalidation *invalidation = palloc(sizeof(*invalidation));
+
+	invalidation->lowest_modified_value = invalidation_form->lowest_modified_value;
+	invalidation->greatest_modified_value = invalidation_form->greatest_modified_value;
+
+	Assert(invalidation->lowest_modified_value <= invalidation->greatest_modified_value);
+	/* this will pick up invalidations where the mat threshold lies inside the
+	 * range. so if we had [1Jan, 20Jan] as the invalidation range and our
+	 * threshold is Jan8. This will pick it up
+	 */
+	if (invalidation->lowest_modified_value < scan_state->materialization_invalidation_threshold)
```

Comment:
are you sure this should be `<` and not `<=`

Reply:
the materialization range is [ completed_threshold, invalidation_threshold) for the single aggr. case.  So it is [completed_threshold, materailziation_inval_threshold) for multiple case and we should pick up only invalidations < materialization_invalidation_threshold. @JLockerman  could you check?

Reply:
`update_materializations` will pin it to be below the new stuff no matter what you pick, so it shouldn't matter that much.

Reply:
and, in fact, since the later code is written with the expectation that it would see all the invalidation, I'd rather we not do any of this checking here, and just use the same logic from `scan_take_invalidation_tuple`, and possibly just use `scan_take_invalidation_tuple` (the structs are layout-compatible, so IIRC it is fine to cast from one to the other)

Reply:
defer for later as an optimization.